pkg/replica/hash.go

```diff
@@ -246,93 +243,141 @@ func (t *SnapshotHashJob) isSnapshotSilentlyCorrupted(checksum string) bool {
 	return false
 }
 
-func GetSnapshotModTime(snapshotName string) (string, error) {
+func GetSnapshotChangeTime(snapshotName string) (string, error) {
 	fileInfo, err := os.Stat(diskutil.GenerateSnapshotDiskName(snapshotName))
 	if err != nil {
 		return "", err
 	}
 
-	return fileInfo.ModTime().String(), nil
+	stat := fileInfo.Sys().(*syscall.Stat_t)
+	return time.Unix(int64(stat.Ctim.Sec), int64(stat.Ctim.Nsec)).String(), nil
 }
 
-func GetSnapshotHashInfoFromXattr(snapshotName string) (*SnapshotXattrHashInfo, error) {
-	xattrSnapshotHashValue := make([]byte, xattrSnapshotHashValueMax)
-	_, err := unix.Getxattr(diskutil.GenerateSnapshotDiskName(snapshotName), xattrSnapshotHashName, xattrSnapshotHashValue)
+func GetSnapshotHashInfoFromChecksumFile(snapshotName string) (*SnapshotXattrHashInfo, error) {
+	dir, err := os.Getwd()
```

Comment:
Why `os.Getwd()` instead of `/var/lib/longhonr/..`?

Reply:
`os.Getwd()` can get the replica dir, which is the working dir of the sync-agent ([codes](https://github.com/longhorn/longhorn-engine/blob/master/app/cmd/replica.go#L156))